### ISSUE TITLE
[Task]: Fix or remove githubPush trigger 

### ISSUE THREAD

**Issue by Abacn** [apache/beam#26561]
### What needs to happen?

After #6097 PostCommit are supposed to run "as soon as new change comes to master." However this is not the case (in my memory). It may be broken long time ago. We may just remove given that we constantly reach GitHub API limit and this probably saves some API call

relevant code path:

https://github.com/apache/beam/blob/fdc489f2f09cc2f2865f50ea8c886a98da2e32fc/.test-infra/jenkins/CommonJobProperties.groovy#L240

https://github.com/apache/beam/blob/fdc489f2f09cc2f2865f50ea8c886a98da2e32fc/.test-infra/jenkins/PostcommitJobBuilder.groovy#L61
 

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner
